web-marketplace/src/legacy-pages/Dashboard/Dashboard.tsx

```diff
@@ -49,6 +64,59 @@ import { useBridgeAvailability } from './hooks/useBridgeAvailabilty';
 import { usePathSection } from './hooks/usePathSection';
 import { useFetchProjectByAdmin } from './MyProjects/hooks/useFetchProjectsByAdmin';
 
+type DashboardNavAccount = AccountOption & {
+  source: 'auth' | 'dao';
+  roleAccountId?: string;
+  roleName?: string;
+  onChainRoleId?: string;
+};
+
+// Legacy IDs cover pre-RBAM assignments still stored in the indexer.
+const legacyRoleIdEntries: Array<[number, string]> = [
+  [1, ROLE_OWNER],
+  [2, ROLE_ADMIN],
+  [3, ROLE_EDITOR],
+  [4, ROLE_AUTHOR],
+  [5, ROLE_VIEWER],
+];
+
+// Current IDs are derived from the hardcoded RBAM role tables.
+const derivedRoleIdEntries: Array<[number, string]> = [
+  [orgRoles.owner.roleId, ROLE_OWNER],
+  [orgRoles.admin.roleId, ROLE_ADMIN],
+  [orgRoles.editor.roleId, ROLE_EDITOR],
+  [orgRoles.viewer.roleId, ROLE_VIEWER],
+  [projectRoles.owner.roleId, ROLE_OWNER],
+  [projectRoles.admin.roleId, ROLE_ADMIN],
+  [projectRoles.editor.roleId, ROLE_EDITOR],
+  [projectRoles.author.roleId, ROLE_AUTHOR],
+  [projectRoles.viewer.roleId, ROLE_VIEWER],
+];
+
+const ON_CHAIN_ROLE_ID_TO_ROLE: Record<string, string> = [
+  ...legacyRoleIdEntries,
+  ...derivedRoleIdEntries,
+].reduce<Record<string, string>>((acc, [roleId, role]) => {
+  acc[String(roleId)] = role;
+  return acc;
+}, {});
```

Comment:
I'm not sure why you need all these mappings
in a `assignment`, you've got both `roleName` and `onChainRoleId`
you can just get the authorization id based on the `roleName` and the authorization name (depends on what should be executed) through the `orgRoles` var I had set up